js-plot/README.md

```diff
@@ -2,22 +2,214 @@
+
+So the bindings look like
+
+```cpp
+#include <emscripten/bind.h>
+#include "newtonraphson.hpp"
+
+using namespace emscripten;
+
+EMSCRIPTEN_BINDINGS(newtonraphson) {
+  class_<NewtonRaphson>("NewtonRaphson")
+    .constructor<double>()
+    .function("solve", &NewtonRaphson::solve)
+    .property("iterations", &NewtonRaphson::iterations)
+    ;
+
+  value_object<Iteration>("Iteration")
+    .field("index", &Iteration::index)
+    .field("x", &Iteration::x)
+    .field("y", &Iteration::y)
+    .field("slope", &Iteration::slope)
+    .field("delta_x", &Iteration::delta_x)
+    ;
+
+  register_vector<Iteration>("vector<Iteration>");
+}
+
+```
+File: _bindings.cpp_.
+
+Same as in previous blog we can compile to a WebAssembly module with Emscripten using `emcc` command
+
+```shell
+emcc \
+  -I. \
+  -o newtonraphson.js \
+  -Oz \
+  -s MODULARIZE=1 \
+  -s EXPORT_NAME=createModule \
+  --bind newtonraphson.cpp bindings.cpp
+
+```
+
+## Vega-lite decleration
 
 TODO Introduce Vega
 
 TODO explain vega-spec.json layer construction
 
-When we visit the web page, we will be greeted by the following plot. We can zoom with mouse wheel and pan by dragging.
+```html
+<html>
+  <head>
+    <script type="text/javascript" src="newtonraphson.js"></script>
+    <script src="https://cdn.jsdelivr.net/npm/vega@5.13.0"></script>
+    <script src="https://cdn.jsdelivr.net/npm/vega-lite@4.13.0"></script>
+    <script src="https://cdn.jsdelivr.net/npm/vega-embed@6.8.0"></script>
+  </head>
+  <body>
+    <div id="plot"></div>
+    <script>
+      createModule().then(async (rootfinding) => {
+        const tolerance = 0.001;
+        const initial_guess = -4;
+        const newtonraphson = new rootfinding.NewtonRaphson(tolerance);
+        newtonraphson.solve(initial_guess);
+        // newtonraphson.iterations is a vector object which not consumeable by Vega
+        // So convert Emscripten vector of objects to JavaScript array of objects
+        const iterations = new Array(
+            newtonraphson.iterations.size()
+        ).fill().map(
+          (_, iteration) => {
+            return newtonraphson.iterations.get(iteration)
+          }
+        );
+        // Open console in DevTools (F12) to see iterations data as a JSON string
+        console.log(JSON.stringify(iterations, null, 2));
+        // Read vega-lite declaration
+        const spec_request = await fetch('vega-spec.json');
+        // TODO decide for spec to be fetched and overwrites or fetched or inlined
+        const spec = await spec_request.json();
+        // ovewrite equation range
+        spec.layer[0].data.sequence.start = initial_guess;
+        spec.layer[0].data.sequence.stop = -1 * initial_guess;
+        // overwrite data
+        spec.layer[2].data.values = iterations;
+        vegaEmbed(document.getElementById("plot"), spec);
+      });
```

Comment:
Do we want to add a HTML table in this PR?

Reply:
no, i think tooltips and output of cli is enough.